FrontMen/fm-hours
# Feature/standby hours

# Changes
Employee can have standby hours toggled on and off
Timesheet now features standby hours if feature is turned on
Timesheets admin view shows stand by hours for all employees, with a label if it is no longer available. Travel expenses also always on (given that toggling it off would hide the history from when it was available). 
Employee cannot see the history of that when it isn't enabled anymore. 

additionally firestore collections moved to an enum

## Related issues
https://github.com/FrontMen/fm-hours/issues/171

## Added

New firestore collection for standby hours
Standby toggler in manage employee view
Standby hours in timesheet both on home and /timesheets

## Changed
Time records service - methods made generic to be able to perform CRUD on both time reports and standby hours

## How to test
1. Login as admin
2. Go to employees and toggle on standby hours
3. Go to home page and see a new row to fill. 
4. Fill it, save it, see it persists. You can also check firestore that it correctly updated
5. Go to /timesheets and check the sheet you filled in - see the new standby area
6. Go to another employee's timesheets, who doesn't have stand by enabled, see a label warning you it isn't available
7. Do the same for an employee without travel enabled

## Screenshots
![image](https://user-images.githubusercontent.com/85111889/126654322-8cc8dded-81ca-42fd-8bba-da1ec5e5a246.png)
![image](https://user-images.githubusercontent.com/85111889/126654378-a5ed1349-e4e1-477c-82e4-0c2183ad9d8e.png)
![image](https://user-images.githubusercontent.com/85111889/126654439-aae0ee92-b338-4151-9fdb-588792ec0654.png)

after rebase with Dean's changes 
![image](https://user-images.githubusercontent.com/85111889/126658440-fd9c5102-7833-41d7-895b-b10c6af821e8.png)

Lasse's enhancement recommandation
![image](https://user-images.githubusercontent.com/85111889/126767226-6e35ab23-5a72-4446-beed-37f4b7a3d2a2.png)
